org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/QuickFixProcessor.java

```diff
@@ -0,0 +1,921 @@
+		case IProblem.UnusedTypeParameter:
+		case IProblem.NullableFieldReference:
+		case IProblem.ConflictingNullAnnotations:
+		case IProblem.ConflictingInheritedNullAnnotations:
+		case IProblem.ExplicitThisParameterNotBelow18:
+		case IProblem.DefaultMethodNotBelow18:
+		case IProblem.StaticInterfaceMethodNotBelow18:
+		case IProblem.LambdaExpressionNotBelow18:
+		case IProblem.MethodReferenceNotBelow18:
+		case IProblem.ConstructorReferenceNotBelow18:
+		case IProblem.IntersectionCastNotBelow18:
+		case IProblem.InvalidUsageOfTypeAnnotations:
+		case IProblem.DuplicateInheritedDefaultMethods:
+		case IProblem.InheritedDefaultMethodConflictsWithOtherInherited:
+		case IProblem.IllegalTypeAnnotationsInStaticMemberAccess:
+		case IProblem.TypeAnnotationAtQualifiedName:
+		case IProblem.NullAnnotationAtQualifyingType:
+		case IProblem.IllegalAnnotationForBaseType:*/
+			return true;
+		default:
+			//return SuppressWarningsSubProcessor.hasSuppressWarningsProposal(cu.getJavaProject(), problemId)
+			//		|| ConfigureProblemSeveritySubProcessor.hasConfigureProblemSeverityProposal(problemId);
+		}
+		return false;
+	}
+
+	private static int moveBack(int offset, int start, String ignoreCharacters, ICompilationUnit cu) {
+		try {
+			IBuffer buf= cu.getBuffer();
+			while (offset >= start) {
+				if (ignoreCharacters.indexOf(buf.getChar(offset - 1)) == -1) {
+					return offset;
+				}
+				offset--;
+			}
+		} catch(JavaModelException e) {
+			// use start
+		}
+		return start;
+	}
+
+
+	public CUCorrectionProposal[] getCorrections(IInvocationContext context, IProblemLocation[] locations) throws CoreException {
+		if (locations == null || locations.length == 0) {
+			return null;
+		}
+
+		HashSet<Integer> handledProblems= new HashSet<>(locations.length);
+		ArrayList<CUCorrectionProposal> resultingCollections = new ArrayList<>();
+		for (int i= 0; i < locations.length; i++) {
+			IProblemLocation curr= locations[i];
+			Integer id= new Integer(curr.getProblemId());
+			if (handledProblems.add(id)) {
+				process(context, curr, resultingCollections);
+			}
+		}
+		return resultingCollections.toArray(new CUCorrectionProposal[resultingCollections.size()]);
+	}
+
+	private void process(IInvocationContext context, IProblemLocation problem,
+			Collection<CUCorrectionProposal> proposals) throws CoreException {
+		int id= problem.getProblemId();
+		if (id == 0) { // no proposals for none-problem locations
+			return;
+		}
+		switch (id) {
+		case IProblem.UnterminatedString:
+			String quoteLabel= CorrectionMessages.JavaCorrectionProcessor_addquote_description;
+			int pos= moveBack(problem.getOffset() + problem.getLength(), problem.getOffset(), "\n\r", context.getCompilationUnit()); //$NON-NLS-1$
+			proposals.add(new ReplaceCorrectionProposal(quoteLabel, context.getCompilationUnit(), pos, 0, "\"", IProposalRelevance.ADD_QUOTE)); //$NON-NLS-1$
+			break;
+		case IProblem.UnusedImport:
+		case IProblem.DuplicateImport:
```

Comment:
proposal ends up with wrong textedit: 
![jun-16-2017 15-50-05](https://user-images.githubusercontent.com/148698/27242530-aaa6ae78-52ab-11e7-9037-8ad1b800884e.gif)

A new line is missing